### DISCUSSION TITLE
Add `docker-compose.yml` with `graph init`

### DISCUSSION THREAD

**Discussion by schmidsi** [graphprotocol/graph-tooling#1599] [Ideas]
In oder to incentivise more local development, I think it would be nice to have a `docker-compose.yml` file automatically generated during `graph init`. I think the [one in the graph-node repo](https://github.com/graphprotocol/graph-node/blob/master/docker/docker-compose.yml) is pretty good.

`dev` and `clean` commands as described here: https://github.com/schmidsi/hackathon-starterkit/blob/main/package.json#L3 might be good.

I think best would be to only add it if it is selected during the `init` dialog. Similar to:

```
✔ Generate ABI and schema types with yarn codegen
```

And enable it by default. Thoughts?

**Comment by saihaj** [comment]
The subset of users I have seen/worked with they have a  smart contract already and having a local `docker-compose` wouldn't be of much value because they likely don't even know how to deploy the contract to the local. Adding `docker-compose` is fine, will convert this to an issue.

**Reply by saihaj** [reply]
https://github.com/graphprotocol/graph-tooling/issues/1602
